### PR TITLE
Enhanced I2C API doc

### PR DESCRIPTION
**Pull request by qiutongs**
Description
-----------

Enhanced I2C API doc

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/common_io/include/iot_i2c.h [hunk 1]
```diff
@@ -247,8 +320,9 @@ int32_t iot_i2c_write_sync( IotI2CHandle_t const pxI2CPeripheral,
  * Partial read might happen, e.g. slave device unable to receive more data.
  * And the number of bytes that have been actually read can be obtained by calling iot_i2c_ioctl.
  *
- * @Note Usually, the address of register needs to be written before calling this function.
- * @Note If eI2CSendNoStopFlag is set and this function returns, whether the actual transaction has been started is undefined. This is board-specific behavior.
+ * @note Usually, the address of register needs to be written before calling this function.
+ * @note If eI2CSendNoStopFlag is set and this function returns, whether the actual transaction has been started is undefined. This is board-specific behavior.
+ * @note Usually, iot_i2c_set_callback is called prior to this in order to get notification when the asynchronous call completes.
```
Comment (vjjunnut) [inline libraries/abstractions/common_io/include/iot_i2c.h]: Should it say "iot_i2c_set_callback() must be called prior to calling this API in order to get notification when the asynchronous operation is completed"

File: libraries/abstractions/common_io/include/iot_i2c.h [hunk 2]
```diff
@@ -287,8 +361,9 @@ int32_t iot_i2c_read_async( IotI2CHandle_t const pxI2CPeripheral,
  * Partial write might happen, e.g. slave device unable to receive more data.
  * And the number of bytes that have been actually written can be obtained by calling iot_i2c_ioctl.
  *
- * @Note Usually, the first byte is treated as the register address and the following bytes are treated as data to be written.
- * @Note If eI2CSendNoStopFlag is set and this function returns, whether the actual transaction has been started is undefined. This is board-specific behavior.
+ * @note Usually, the first byte is treated as the register address and the following bytes are treated as data to be written.
+ * @note If eI2CSendNoStopFlag is set and this function returns, whether the actual transaction has been started is undefined. This is board-specific behavior.
+ * @note Usually, iot_i2c_set_callback is called prior to this in order to get notification when the asynchronous call completes.
```
Comment (vjjunnut) [inline libraries/abstractions/common_io/include/iot_i2c.h]: Same as #325

File: libraries/abstractions/common_io/include/iot_i2c.h (hunk 1 above)
Comment (qiutongs) [reply libraries/abstractions/common_io/include/iot_i2c.h]: You want set_callback mandatory? I was thinking this is optional, based on what user needs.

Comment (vjjunnut) [reply libraries/abstractions/common_io/include/iot_i2c.h]: No. Looks like even my comments are not clear..:-)
I meant - callback must be set for notifications to happen

Comment (qiutongs) [reply libraries/abstractions/common_io/include/iot_i2c.h]: Ah, my bad. OK, I will rephrase it.

**Pull request merged**